extensions/wikia/FlowTracking/scripts/createNewPageFlows.js

```diff
@@ -8,25 +8,43 @@ define(
 			articleId = mw.config.get('wgArticleId');
 
 		function trackOnEditPageLoad(editor) {
+			var qs = new QueryString(),
+				// 'flow' is the parameter passed in the url if user has started a flow already
+				flowParam = qs.getVal('flow', false),
+				tracked = qs.getVal('tracked', false);
+
+			// Do not track if the step was tracked already
+			if (tracked) {
+				return;
+			}
+
 			// Track only creating articles (wgArticleId=0) from namespace 0 (Main)
 			// IMPORTANT: on Special:CreatePage even after providing article title the namespace is set to -1 (Special Page)
 			if (namespaceId === 0 && articleId === 0) {
-				var qs = new QueryString(),
-				// 'flow' is the parameter passed in the url if user has started a flow already
-					flowParam = qs.getVal('flow', false);
-
 				if (flowParam || document.referrer) {
-					//TODO: track middle step for other flows
+					flowTrack.trackFlowStep(flowParam, {editor: editor});
 				} else {
 					flowTrack.beginFlow(flowTrack.flows.CREATE_PAGE_DIRECT_URL, {editor: editor});
 					qs.setVal('flow', flowTrack.flows.CREATE_PAGE_DIRECT_URL);
 					window.history.replaceState({}, '', qs.toString());
```

Comment:
We can remove this line, as this will happen anyway at the end of the function. Are you fine with that?